stdlib/public/SDK/Foundation/NSError.swift

```diff
@@ -366,93 +366,59 @@ public func _bridgeNSErrorToError<
-  public static func ==(lhs: Self, rhs: Self) -> Bool {
-    return lhs.rawValue == rhs.rawValue
-  }
+extension _BridgedNSError {
+  public var _domain: String { return Self._nsErrorDomain }
 }
 
-public extension __BridgedNSError 
-    where Self: RawRepresentable, Self.RawValue: SignedInteger {
-  public var _domain: String { return Self._nsErrorDomain }
+extension _BridgedNSError where Self.RawValue: SignedInteger {
   public var _code: Int { return Int(rawValue) }
 
-  public init?(rawValue: RawValue) {
-    self = unsafeBitCast(rawValue, to: Self.self)
-  }
-
   public init?(_bridgedNSError: NSError) {
     if _bridgedNSError.domain != Self._nsErrorDomain {
       return nil
     }
 
-    self.init(rawValue: RawValue(IntMax(_bridgedNSError.code)))
+    self.init(rawValue: RawValue(_bridgedNSError.code))
   }
 
   public var hashValue: Int { return _code }
 }
 
-// Allow two bridged NSError types to be compared.
-extension __BridgedNSError
-    where Self: RawRepresentable, Self.RawValue: UnsignedInteger {
-  public static func ==(lhs: Self, rhs: Self) -> Bool {
-    return lhs.rawValue == rhs.rawValue
-  }
-}
-
-public extension __BridgedNSError
-    where Self: RawRepresentable, Self.RawValue: UnsignedInteger {
-  public var _domain: String { return Self._nsErrorDomain }
+extension _BridgedNSError where Self.RawValue: UnsignedInteger {
   public var _code: Int {
     return Int(bitPattern: UInt(rawValue))
   }
 
-  public init?(rawValue: RawValue) {
-    self = unsafeBitCast(rawValue, to: Self.self)
-  }
-
   public init?(_bridgedNSError: NSError) {
     if _bridgedNSError.domain != Self._nsErrorDomain {
       return nil
     }
 
-    self.init(rawValue: RawValue(UIntMax(UInt(_bridgedNSError.code))))
+    self.init(rawValue: RawValue(UInt(bitPattern: _bridgedNSError.code)))
   }
 
   public var hashValue: Int { return _code }
 }
 
-/// Describes a raw representable type that is bridged to a particular
-/// NSError domain.
-///
-/// This protocol is used primarily to generate the conformance to
-/// _ObjectiveCBridgeableError for such an enum.
-public protocol _BridgedNSError : __BridgedNSError,
-                                  RawRepresentable,
-                                  _ObjectiveCBridgeableError,
-                                  Hashable {
-  /// The NSError domain to which this type is bridged.
-  static var _nsErrorDomain: String { get }
-}
-
 /// Describes a bridged error that stores the underlying NSError, so
 /// it can be queried.
 public protocol _BridgedStoredNSError :
-     __BridgedNSError, _ObjectiveCBridgeableError, CustomNSError,
-     Hashable {
+     _ObjectiveCBridgeableError, CustomNSError, Hashable {
   /// The type of an error code.
-  associatedtype Code: _ErrorCodeProtocol
-
-  /// The error code for the given error.
-  var code: Code { get }
+  associatedtype Code: _ErrorCodeProtocol, RawRepresentable
+  where Code.RawValue: FixedWidthInteger
```

Comment:
This was the last thing I couldn't update. These constraints ought to be on _ErrorCodeProtocol itself, because there'll never be any _ErrorCodeProtocol types that don't match, but the compiler crashes if I try to do that when we actually go to use the conformance. I'll try to come up with a reduced case that doesn't depend on the importer.

Reply:
Okay, no luck with a reduction, but that's because it's the ClangImporter code I mentioned above that's at fault. Clearly it needs more actual attention and not a quick fix.